 Fix Dds switch issue

During data call setup in progress on Dds sub, if Dds is switched,
data disconnected indication is not received immediately,
due to which PS attach is delayed.
Fix is to notify data disconnect indication if data call is not in
connected state on Dds sub.

diff --git a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
index 000dce3..60d02a0 100644
--- a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
+++ b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
@@ -3261,8 +3261,8 @@ protected void notifyAllDataDisconnected() {
     public void registerForAllDataDisconnected(Handler h, int what, Object obj) {
         mAllDataDisconnectedRegistrants.addUnique(h, what, obj);
 
-        if (isDisconnected()) {
-            log("notify All Data Disconnected");
+        if (!isConnected()) {
+            log("No Apncontext is connected, notify All Data Disconnected");
             notifyAllDataDisconnected();
         }
     }
